tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -951,7 +971,7 @@ def __init__(self,
                input_saved_model_dir=None,
                input_saved_model_tags=None,
                input_saved_model_signature_key=None,
-               conversion_params=DEFAULT_TRT_CONVERSION_PARAMS):
+               conversion_params=TrtConversionParams()):
```

Comment:
This still uses a mutable object as the default value of the function. That's quite dangerous as this object can be inadvertently overwritten.

tf-api-owners prefer keeping TrtConversionParams immutable and using a replace method or, if you do want it mutable, pass None here and create a default instance in the body of the function

See https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html for a reference on why mutable default arguments are dangerous.

Reply:
> This still uses a mutable object as the default value of the function. That's quite dangerous as this object can be inadvertently overwritten.
> 
> tf-api-owners prefer keeping TrtConversionParams immutable and using a replace method or, if you do want it mutable, pass None here and create a default instance in the body of the function
> 
> See https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html for a reference on why mutable default arguments are dangerous.

@alextp Thanks for the link. I made `TrtConversionParams` immutable by overriding `__setattr__`. Please let me know if that's not enough. 

We can't change the default value to None due to backward compatibility.
We don't really need `TrtConversionParams` to be mutable.

Reply:
> We can't change the default value to None due to backward compatibility.

I may be missing some Python subtleties, but can you default it to `None` and then within the function do

```
if conversion_params is None:
  conversion_params = TrtConversionParams()
```
?

Reply:
@sanjoy Thanks, that would actually work. I fixed that.